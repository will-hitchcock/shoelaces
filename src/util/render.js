cosnt render = (comp, props) => {
  return (props) => {
    let newProps = {}
    if (comp.defaultProps) {
      new props = (...comp.defaultProps, ...props)
    } else {
      newProps = props
    }
    return comp(newProps)
  }
}
