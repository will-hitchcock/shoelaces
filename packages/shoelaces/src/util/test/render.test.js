import render from '../render';

describe('render', () => {
  const testComp = ({ prop1, prop2 }) => ({ prop1, prop2 });

  testComp.defaultProps = { prop2: 'prop2' };

  it('should merge default props and call comp', () => {
    expect(render(testComp)({ prop1: 'prop1' })).toMatchSnapshot();
  });

  it('should merge default props and call comp', () => {
    expect(render(testComp)({ prop1: 'prop1', prop2: false })).toMatchSnapshot();
  });

  it('should only validate when the propType is a function', () => {
    testComp.propTypes = { prop1: false };
    expect(render(testComp)({ prop1: 'prop1' })).toMatchSnapshot();
  });
});
