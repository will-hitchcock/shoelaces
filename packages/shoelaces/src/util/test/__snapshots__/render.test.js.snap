// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`render should merge default props and call comp 1`] = `
Object {
  "prop1": "prop1",
  "prop2": "prop2",
}
`;

exports[`render should merge default props and call comp 2`] = `
Object {
  "prop1": "prop1",
  "prop2": false,
}
`;

exports[`render should only validate when the propType is a function 1`] = `
Object {
  "prop1": "prop1",
  "prop2": "prop2",
}
`;
